lib/Onnxifi/base.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "base.h"
+
+namespace glow {
+namespace onnxifi {
+
+bool Event::signal() {
+    std::lock_guard<std::mutex> guard(mutex_);
+    if (fired_) {
+      return false;
+    }
+    fired_ = true;
+    cond_.notify_all();
```

Comment:
You might want to release the mutex_ before calling notify_all.  It's a small optimization, but this way the waiting thread can immediately start once it receives the notify, rather than waiting until the mutex release.